hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ConfigCheck.java

```diff
@@ -179,7 +171,8 @@ public int getId() {
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeUTF(groupName);
-        out.writeUTF(groupPassword);
+        //TODO @tkountis remove in 4.0
+        out.writeUTF(EMPTY_PWD);
```

Comment:
I think conditionally sending the group password has value because `ConfigCheck` is used in multicast messages throughout the lifetime of the member and in split-brain join messages, not just until cluster is initially joined:
* Assuming an existing 3.8 EE cluster
* Rolling upgrade each member to 3.9 EE members (`UNKNOWN < 3.9` --> 3.9 member still sends out pwd and is allowed to join the 3.8 EE cluster)
* As long as cluster version is still `3.8`, the split-brain join message sent out from a 3.9 member will include the group password and will be allowed to join back to the 3.8 cluster.
* Once all members are upgraded, upgrade cluster version to 3.9 --> from now on multicast & split-brain join messages have a proper `ConfigCheck` with an empty password.

Otherwise, if in 3.9 the pwd is dropped, one would need to upgrade first 3.8 -> 3.8.2+ and then move on to 3.9. wdyt?